<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)" />
  <arg name="portname" default="/dev/vio_imu" />
  <arg name="profiler" default="$(optenv PROFILER false)" />
  <arg name="verbose" default="true" />
  
  <arg name="custom_config" default="" />

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_vio_imu_manager" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg UAV_NAME)">

    <!-- launch the nodelet -->
    <node pkg="nodelet" type="nodelet" name="vio_imu" args="$(arg nodelet) vio_imu/VioImu $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <param name="uav_name" value="$(arg UAV_NAME)"/>

      <rosparam file="$(find mrs_serial)/config/imu_default.yaml" command="load" />
      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />

      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="portname" value="$(arg portname)"/>
      <param name="verbose" value="$(arg verbose)"/>

      <!-- Publishers -->
      <remap from="~profiler" to="profiler" />
      <remap from="~baca_protocol_out" to="~received_message" />

        <!-- Subscribers -->
      <remap from="~baca_protocol_in" to="~send_message" />
      <remap from="~raw_in" to="~send_raw_message" />

    </node>

  </group>

</launch>
